field/src/packed.rs

```diff
@@ -4,13 +4,77 @@ use core::{mem, slice};
 use crate::field::Field;
 use crate::AbstractField;
 
+/// A trait to constrain types that can be packed into a packed value.
+///
+/// The `Packable` allows us to specify implementations for potentially conflicting types.
```

Comment:
Some relatedly, I've been pondering possibly bringing back `TrivialPacking` (as well as `TrivialExtension`) which existed in a very old version of Plonky3. It would lead to some extra type wrappers, but might be nice for avoiding conflicts with blanket impls like you ran into here.